protocols/kad/src/behaviour.rs

```diff
@@ -1744,19 +1799,22 @@ where
                                     );
                                 }
                             }
-                        } else if quorum.get() == 1 {
-                            // It is a "standard" Kademlia query, for which the
-                            // closest node to the key that did *not* return the
-                            // value is tracked in order to cache the record on
-                            // that node if the query turns out to be successful.
-                            let source_key = kbucket::Key::from(source);
-                            if let Some(cache_key) = cache_at {
-                                let key = kbucket::Key::new(key.clone());
-                                if source_key.distance(&key) < cache_key.distance(&key) {
+                        } else {
+                            no_record.push(source);
+                            if quorum.get() == 1 {
```

Comment:
> I'm not sure if it would be worthwhile to be able to configure (i.e. enable/disable) the default behaviour of write-back caching upon lookups with a quorum of 1. Maybe some users would like full control over that behaviour by disabling it and doing any such writes themselves via Kademlia::put_record_to. Feedback appreciated.

I am fine with either solution. That said I do tend towards not treating _quorum of 1_ as a special case. Instead I suggest to always require the user to call `put_record_to` in order for _write back caching_ to close nodes without the record to happen. Not having  a special case for _quorum of 1_ seems more consistent to me, despite the fact that it is not perfectly inline with the paper.

I would deem an additional note on the `GetRecordOk` structure, informing the user of their responsibility, as well as an entry in the changelog enough. We could as well enforce this at the type level, requiring the collection of peers returned in `GetRecordOk::no_record` to be used, just like `std::result::Result`.

What do you think @romanb?